app/services/post_alerter.rb

```diff
@@ -319,32 +319,33 @@ def create_notification(user, type, post, opts = {})
       ).exists?
     end
 
-    # Don't notify the same user about the same notification on the same post
-    existing_notification = user.notifications
+    existing_notifications = user.notifications
       .order("notifications.id DESC")
-      .find_by(
+      .where(
         topic_id: post.topic_id,
-        post_number: post.post_number,
-        notification_type: type
-      )
+        post_number: post.post_number
+      ).limit(10)
+
+    # Don't notify the same user about the same type of notification on the same post
+    existing_notification_of_same_type = existing_notifications.find { |n| n.notification_type == type }
```

Comment:
I don't understand how filtering in Ruby is any different than filtering in SQL?

Reply:
It's one query instead of two, so it should be more performant?

Reply:
My bad, I didn't see you were reusing this at the end of the method.